src/Type.h

```diff
@@ -20,81 +20,92 @@ struct Expr;
  * larger than one). Front-end code shouldn't use vector
  * types. Instead vectorize a function. */
 struct Type {
-    /** The basic type code: signed integer, unsigned integer, or floating point.
-     *
-     * Note that TypeCode is guaranteed to have values identical to those of
-     * halide_type_code_t (HalideRuntime.h), but exists as a separate typedef
-     * to preserve source code compatibility.
-     */
-    enum TypeCode {
-        Int = halide_type_int,
-        UInt = halide_type_uint,
-        Float = halide_type_float,
-        Handle = halide_type_handle
-    } code;
-
-    /** The number of bits of precision of a single scalar value of this type. */
-    int bits;
+  private:
+    halide_type_t type;
 
-    /** The number of bytes required to store a single scalar value of this type. Ignores vector width. */
-    int bytes() const {return (bits + 7) / 8;}
+  public:
+    /** Aliases for halide_type_code_t values for legacy compatibility
+     * and to match the Halide internal C++ style. */
+    // @{
+    static constexpr halide_type_code_t Int = halide_type_int;
+    static constexpr halide_type_code_t UInt = halide_type_uint;
+    static constexpr halide_type_code_t Float = halide_type_float;
+    static constexpr halide_type_code_t Handle = halide_type_handle;
+    // @}
 
-    /** How many elements (if a vector type). Should be 1 for scalar types. */
-    int width;
+    /** The number of bytes required to store a single scalar value of this type. Ignores vector width. */
+    int bytes() const {return (bits() + 7) / 8;}
```

Comment:
Likely "yes," but changing existing functionality in Type.h to use unsigned types is going to require either ignoring type sign mistmatches or fixing a lot of code. I opened https://github.com/halide/Halide/issues/979 to make an umbrella task to solve this in the broad sense and I made the version of bytes on halide_type_t return size_t so new uses get the right thing. I'd rather not tackle fixing the entire issue as part of this PR however.